src/transforms/gate/config.rs

```diff
@@ -0,0 +1,96 @@
+use vector_lib::config::{clone_input_definitions, LogNamespace};
+use vector_lib::configurable::configurable_component;
+
+use crate::{
+    conditions::AnyCondition,
+    config::{
+        DataType, GenerateConfig, Input, OutputId, TransformConfig, TransformContext,
+        TransformOutput,
+    },
+    schema,
+    transforms::Transform,
+};
+
+use super::transform::Gate;
+
+
+/// Configuration for the `gate` transform.
+#[configurable_component(transform(
+"gate",
+"Open or close an event stream based on supplied criteria"
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct GateConfig {
+    /// A logical condition used to pass events through without gating.
+    pub pass_when: Option<AnyCondition>,
+
+    /// A logical condition used to open the gate.
+    pub open_when: Option<AnyCondition>,
+
+    /// A logical condition used to close the gate.
+    pub close_when: Option<AnyCondition>,
+
+    /// Maximum number of events to keep in the buffer.
+    pub max_events: Option<usize>,
+
+    /// Automatically close the gate after the buffer has been flushed.
+    pub auto_close: Option<bool>,
```

Comment:
Would we want to disallow `auto_close` and `close_when` being specified at the same time?

It also seems like `auto_close` might be the equivalent of `close_when: "true"` but that may not terribly discoverable behavior. We could treat the absence of `close_when` as auto-close 🤔